### DISCUSSION TITLE
Header action shortcuts not visible enough

### DISCUSSION THREAD

**Discussion by guillenotfound** [ibhagwan/fzf-lua#1620] [Ideas]
So I've tried doing a few searches [here](https://github.com/ibhagwan/fzf-lua/blob/73e7f5f93b65028b9fc94a27a4c80611ff8ba2a0/doc/fzf-lua-opts.txt) to see if I can find a way to get access to the header (mappings):
<img width="519" alt="image" src="https://github.com/user-attachments/assets/385e9ac5-3d13-414b-86a4-c4e3560db051" />

The problem is that at least for me they will trim so it's not very useful and I could not find a way to show the full header... With telescope you can hit one of these two and it will show you all the possible combinations:
<img width="601" alt="image" src="https://github.com/user-attachments/assets/127afb65-9f2c-4c3e-8010-f88fbcef2120" />

To split vertically, horizontally, ctrl+g,.... I think it would be nice because I feel like I'm missing some mappings so I find my self reading on reddit to see how people are using the plugin lol

I've been using this for a week and I'm impressed, such a piece of art @ibhagwan ! Congrats for making such an amazing tool ;)

**Comment by ibhagwan** [comment]
> To split vertically, horizontally, ctrl+g,.... I think it would be nice because I feel like I'm missing some mappings so I find my self reading on reddit to see how people are using the plugin lol

You can press F1 to see help for all the mappings (customizable):
![IMG_2823](https://github.com/user-attachments/assets/7f788981-0fc5-426f-a34a-9921bfd54cc5)

If you run `:FzfLua profiles` and select telescope, you can toggle help with ctrl-/ like telescope.

Also play with the F4/F5/F6 to hide / rotate the preview, it should help in this case.

> I've been using this for a week and I'm impressed, such a piece of art @ibhagwan ! Congrats for making such an amazing tool ;)

Ty for the kind words friend :)

**Reply by ibhagwan** [reply]
#1579 

Some more info regarding line wrap, ended up not affecting the header but still

**Reply by guillenotfound** [reply]
Ohh exactly what I was looking for, thank you so much!

Something else I'm missing is cycling, like when getting to the end of the list of results and hitting `ctrl+n` I'd like it to go to the first result, same but the oposite when using `ctrl+p`. Is there also a setting for that?

**Reply by ibhagwan** [reply]
> Ohh exactly what I was looking for, thank you so much!
> 
> Something else I'm missing is cycling, like when getting to the end of the list of results and hitting `ctrl+n` I'd like it to go to the first result, same but the oposite when using `ctrl+p`. Is there also a setting for that?

fzf_opts .--cycle=true

**Reply by guillenotfound** [reply]
Ohh I see it has been answered multiple times, my bad for not looking into it with more detail: https://github.com/ibhagwan/fzf-lua/issues/861 https://github.com/ibhagwan/fzf-lua/issues/1564 (at least)

Would it make sense to have this in the FAQ? https://github.com/ibhagwan/fzf-lua/wiki

I'm going to go check it in detail but so far I have the setup I'm used to, thanks a lot again! <3

**Reply by ibhagwan** [reply]
> Ohh I see it has been answered multiple times, my bad for not looking into it with more detail: #861 #1564 (at least)
> 
> Would it make sense to have this in the FAQ? https://github.com/ibhagwan/fzf-lua/wiki
> 
> I'm going to go check it in detail but so far I have the setup I'm used to, thanks a lot again! <3

No worries, I answer on the go sometimes so the min answer works too,

Feel free to edit the Wiki, it’s open edit :)

**Reply by guillenotfound** [reply]
Done, thanks again!

**Reply by ibhagwan** [reply]
> Done, thanks again!

Ty @guillenotfound!

**Comment by minusfive** [comment]
@ibhagwan suggestion: instead of making interactive headers the default, what about making a single, simple static `<F1> for help` header? I feel like the interactive header isn't really useful as is since it's typically truncated after the first keymap, and the related code is therefore perhaps doing a bit too much without providing much value.

Happy to take a stab at a PR if you think this is a good idea? For reference this is what I have set up for mine:

![Screenshot 2024-12-21 at 21 15 42](https://github.com/user-attachments/assets/65e1fdeb-7b09-4296-9694-25d070a29741)


Note I'm using 3 keymaps because I'm experimenting to find the most natural for me. I use a corne keyboard due to RSI and my fn keys are on the 3rd layer 'cause I rarely use them, so the default F1 is a bit cumbersome for me.

Here's my code if anyone is interested:

```lua
function(_, opts)
  local fzf_utils = require("fzf-lua.utils")
  local fzf_highlights = require("fzf-lua.config").defaults.__HLS

  LazyVim.merge(opts, {
    winopts = {
      -- border = "none",
      height = 0.5,
      width = 1,
      row = 1,
      col = 0,
    },
    keymap = {
      builtin = {
        true,
        ["<C-h>"] = "toggle-help",
        ["<C-/>"] = "toggle-help",
      },
    },
    defaults = {
      RIPGREP_CONFIG_PATH = vim.env.RIPGREP_CONFIG_PATH,
      header = string.format(
        "%s%s%s%s%s%s%s\n\n",
        fzf_utils.ansi_from_hl(fzf_highlights.header_bind, "<C-h>"),
        fzf_utils.ansi_from_hl(fzf_highlights.header_text, ", "),
        fzf_utils.ansi_from_hl(fzf_highlights.header_bind, "<C-/>"),
        fzf_utils.ansi_from_hl(fzf_highlights.header_text, " or "),
        fzf_utils.ansi_from_hl(fzf_highlights.header_bind, "<F1>"),
        fzf_utils.ansi_from_hl(fzf_highlights.header_text, " for "),
        fzf_utils.ansi_from_hl(fzf_highlights.header_bind, "help")
      ),
    },
    grep = {
      rg_glob = true,
    },
  })
end
```

**Reply by ibhagwan** [reply]
> @ibhagwan suggestion: instead of making interactive headers the default, what about making a single, simple static <F1> for help header? I feel like the interactive header isn't really useful as is since it's typically truncated after the fist keymap, and the related code is therefore perhaps doing a bit too much without providing much value.
> Happy to take a stab at a PR if you think this is a good idea? For reference this is what I have set up for mine:

I agree with there’s room for improvement but my idea also be to toggle the header inside fzf with a `transform` bind which fzf added recently, it’s much more flexible and we can show the headers inside fzf instead of relying on the help screen, obviously this all has to be customizable as every user has their own preference.

Perhaps I’ll tackle this sometime soon but unsure yet, if you wanna experiment with a PR go ahead but it would have to be customizable, scalable and not opinionated.
